src/rules/valid-mock-module-path.ts

```diff
@@ -0,0 +1,121 @@
+import { statSync } from 'fs';
+import path from 'path';
+import { AST_NODE_TYPES, type TSESTree } from '@typescript-eslint/utils';
+import {
+  createRule,
+  findModuleName,
+  getAccessorValue,
+  isSupportedAccessor,
+  isTypeOfJestFnCall,
+} from './utils';
+
+export default createRule<
+  [
+    Partial<{
+      moduleFileExtensions: readonly string[];
+    }>,
+  ],
+  'invalidMockModulePath'
+>({
+  name: __filename,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow mocking of non-existing module path',
+    },
+    messages: {
+      invalidMockModulePath: 'Module path {{ moduleName }} does not exist',
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          moduleFileExtensions: {
+            type: 'array',
+            items: { type: 'string' },
+            additionalItems: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      moduleFileExtensions: ['.js', '.ts', '.tsx', '.jsx', '.json'],
+    },
+  ],
+  create(
+    context,
+    [{ moduleFileExtensions = ['.js', '.ts', '.tsx', '.jsx', '.json'] }],
```

Comment:
can we add a few more tests for this option - we should have one for `.json`, and it would be good to have one that adds e.g. `.css` which'll serve to show both arbitrary extensions work and that it replaces rather than extends the default

(I know you've already got at least one test covering the latter, but its still nice to have another)